File: MarketFlutter/lib/models/stock.dart

import 'package:market/models/offer_type.dart';

class Stock {
  int id;
  int offerTypeId;
  String title;
  String sellerId;
  double quantity;
  OfferType offerType;
  // Constructor
  Stock({required this.id, required this.title, required this.sellerId, required this.quantity, required this.offerTypeId, required this.offerType});

  factory Stock.fromJson(Map<String, dynamic> json) {
    Stock res = Stock(
      id: json['id'] as int,
      title: json['title'] as String,
      sellerId: json['sellerId'] as String,
      offerTypeId: json['offerTypeId'] as int,
      quantity: json['quantity']+.0,
      offerType: OfferType.fromJson(json['offerType']),
    );
    return res;
  }

  // Method to convert User instance to a JSON map
  Map<String, dynamic> toJson() {
    //TODO: Add all types when fix api
    return {
      'id': id,
      'title': title,
      'sellerId': sellerId,
      'quantity': quantity,
      'offerTypeId': offerTypeId,
      'offerType': offerType.toJson(),
    };
  }
}

File: API/MarketAPI/Data/Models/Stock.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MarketAPI.Data.Models
{
    public class Stock
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public string Title { get; set; }
        [Required]
        [ForeignKey(nameof(OfferType))]
        public int OfferTypeId { get; set; }
        public OfferType OfferType { get; set; }
        
        [Required]
        [ForeignKey(nameof(User))]
        public Guid SellerId { get; set; }
        public User Seller { get; set; }

        public List<Offer> Offers { get; set; }
        
        [Required]
        public decimal Quantity { get; set; }
    }
}
